services/tone_analyzer/v3.js

```diff
@@ -0,0 +1,148 @@
+  // service credentials. This separation from sUsername and username to allow 
+  // the end user to modify the node credentials when the service is not bound.
+  // Otherwise, once set username would never get reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
+
+  // Not ever used, and codeacy complains about it.
+
+  var username, password, sUsername, sPassword;
+
+  var service = cfenv.getAppEnv().getServiceCreds(/tone analyzer/i)
+
+  if (service) {
+    sUsername = service.username;
+    sPassword = service.password;
+  }
+
+  // Node RED Admin - fetch and set vcap services
+  RED.httpAdmin.get('/watson-tone-analyzer/vcap', function (req, res) {
+    res.json(service ? {bound_service: true} : null);
+  });
+
+
+  // This is the Tone Analyzer Node. 
+  function Node (config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+
+    // added in this flag as codacy was complaing about the 
+    // cyclomatic complexity. Essentially the number of exit points. With
+    // this flag we can reduce that do two exits. One if there is an
+    // error detected and the second for normal processing
+    var errorDetected = false;
+    var isBuffer = false;
+
+    // Invoked whenb the node has received an input as part of a flow.
+    this.on('input', function (msg) {
+      var message = '';
+
+      // Credentials are needed for each of the modes.
+      username = sUsername || this.credentials.username;
+      password = sPassword || this.credentials.password;      
+
+      if (!username || !password) {
+        message = 'Missing Tone Analyzer service credentials';
+        errorDetected = true;
+      }
+
+      if (!msg.payload) {
+        message = 'Missing property: msg.payload';
+        errorDetected = true;
+      } else {
+        var hasJSONmethod = (typeof msg.payload.toJSON === 'function') ;
+
+        if (hasJSONmethod === true) {
+          if (msg.payload.toJSON().type === 'Buffer') {
+            isBuffer = true;
+          }      
+        }      
+      }
+
+      // Payload (text to be analysed) must be a string (content is either raw string or Buffer)
+      if (typeof msg.payload !== 'string' &&  isBuffer !== true) {
+        errorDetected = true;
+        message = 'The payload must be either a string or a Buffer';
+      }
+
+      if (errorDetected) {
+        node.status({fill:'red', shape:'dot', text:message}); 
+        node.error(message, msg);
+        return;
+      }
+
+      var tones = msg.tones || config.tones;
+      var sentences = msg.sentences || config.sentences;
+      var contentType = msg.contentType || config.contentType
+
+      //var taSettings = {
+      //  'username' : username, 
+      //  'password' : password,
+      //  'tones' : tones,
+      //  'sentences' : sentences,
+      //  'contentType' : contentType
+      //};
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected space or tab after "//" in comment.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2651883225/issues/source?bid=3225115&fileBranchId=3324990#l106)